src/dune/expander.ml

```diff
@@ -1,5 +1,12 @@
 open Stdune
 
+type expanded =
+  | Static of Value.t list
+  | Dynamic of Pform.Expansion.t
+  | Error of User_message.t
```

Comment:
What about:

```
type expanded = Static of .. | Dynamic of ..
```

and then use:

```
(expanded, User_message.t) Result.t
```

After working a lot of Lwt, I'm not a huge of fan inlining the error case in any type.

Reply:
OK, done.